File: launchdarkly/java/src/main/java/oseg/launchdarkly_examples/SearchContextsExample.java
package oseg.launchdarkly_examples;

import com.launchdarkly.client.ApiException;
import com.launchdarkly.client.Configuration;
import com.launchdarkly.client.api.*;
import com.launchdarkly.client.auth.*;
import com.launchdarkly.client.JSON;
import com.launchdarkly.client.model.*;

import java.io.File;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.OffsetDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public class SearchContextsExample
{
    public static void main(String[] args)
    {
        var config = Configuration.getDefaultApiClient();
        ((ApiKeyAuth) config.getAuthentication("ApiKey")).setApiKey("YOUR_API_KEY");

        var contextSearch = new ContextSearch();
        contextSearch.filter("*.name startsWith Jo,kind anyOf [\"user\",\"organization\"]");
        contextSearch.sort("-ts");
        contextSearch.limit(10);
        contextSearch.continuationToken("QAGFKH1313KUGI2351");

        try
        {
            var response = new ContextsApi(config).searchContexts(
                "projectKey_string", // projectKey
                "environmentKey_string", // environmentKey
                contextSearch,
                null, // limit
                null, // continuationToken
                null, // sort
                null, // filter
                null // includeTotalCount
            );

            System.out.println(response);
        } catch (ApiException e) {
            System.err.println("Exception when calling ContextsApi#searchContexts");
            System.err.println("Status code: " + e.getCode());
            System.err.println("Reason: " + e.getResponseBody());
            System.err.println("Response headers: " + e.getResponseHeaders());
            e.printStackTrace();
        }
    }
}

File: launchdarkly/kotlin/src/main/kotlin/oseg/launchdarkly_examples/SearchContextsExample.kt
package oseg.launchdarkly_examples

import com.launchdarkly.client.infrastructure.*
import com.launchdarkly.client.apis.*
import com.launchdarkly.client.models.*

import java.io.File
import java.time.LocalDate
import java.time.OffsetDateTime
import kotlin.collections.ArrayList
import kotlin.collections.List
import kotlin.collections.Map
import com.squareup.moshi.adapter

@ExperimentalStdlibApi
class SearchContextsExample
{
    fun searchContexts()
    {
        ApiClient.apiKey["ApiKey"] = "YOUR_API_KEY"

        val contextSearch = ContextSearch(
            filter = "*.name startsWith Jo,kind anyOf [\"user\",\"organization\"]",
            sort = "-ts",
            limit = 10,
            continuationToken = "QAGFKH1313KUGI2351",
        )

        try
        {
            val response = ContextsApi().searchContexts(
                projectKey = null,
                environmentKey = null,
                contextSearch = contextSearch,
                limit = null,
                continuationToken = null,
                sort = null,
                filter = null,
                includeTotalCount = null,
            )

            println(response)
        } catch (e: ClientException) {
            println("4xx response calling ContextsApi#searchContexts")
            e.printStackTrace()
        } catch (e: ServerException) {
            println("5xx response calling ContextsApi#searchContexts")
            e.printStackTrace()
        }
    }
}
